### ISSUE TITLE
Unexpected token ":" when defining object literal's properties

### ISSUE THREAD

**Issue by Gelio** [eslint/eslint#7259]
**Tell us about your environment**

* **ESLint Version:** v2.13.1
* **Node Version:** v6.5.0
* **npm Version:** 3.10.7

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
``` json
{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "script",
        "ecmaFeatures": {
            "impliedStrict": true
        }
    },
    "env": {
        "node": true,
        "jasmine": true,
        "es6": true
    },
    "globals": {
        "sails": true,
        "Promise": true
    },
    "rules": {
        "no-cond-assign": "error",
        "no-dupe-args": "error",
        "no-dupe-keys": "error",
        "no-duplicate-case": "error",
        "no-empty": "warn",
        "no-extra-semi": "warn",
        "no-sparse-arrays": "warn",
        "no-unreachable": "warn",
        "valid-typeof": "error",

        "curly": ["warn", "all"],
        "dot-location": ["warn", "property"],
        "dot-notation": "warn",
        "eqeqeq": "error",
        "no-empty-function": "warn",
        "no-eval": "error",
        "no-implied-eval": "error",
        "no-multi-spaces": "warn",
        "no-octal": "error",
        "no-throw-literal": "error",
        "radix": "error",

        "no-shadow": "error",
        "no-undef": "error",
        "no-unused-vars": "warn",

        "block-spacing": "warn",
        "brace-style": ["warn", "1tbs"],
        "camelcase": "warn",
        "comma-spacing": "warn",
        "eol-last": "error",
        "indent": ["warn", 2, {
            "SwitchCase": 1
        }],
        "max-len": ["warn", 100],
        "new-parens": "error",
        "no-mixed-spaces-and-tabs": "warn",
        "no-trailing-spaces": "warn",
        "quotes": ["warn", "single", {
            "allowTemplateLiterals": true
        }],
        "semi": ["error", "always"],

        "no-const-assign": "error",
        "no-dupe-class-members": "error",
        "no-this-before-super": "error",
        "no-var": "error"
    }
}
```

**What did you do? Please include the actual source code causing the issue.**
I get linting errors when defining properties of object literals:
![](https://i.gyazo.com/8f9a655ebe4cebffa6d1cae2d01e8719.png)
Here's the code:
``` javascript
module.exports.auth = {
  bcrypt: {
    rounds: 8
  },
  entryTokenLength: 16
};
```
It isn't an issue with my IDE (Visual Studio Code, btw.), because I ran eslint against it from the terminal and the result was the same (see below).

**What did you expect to happen?**
I don't think it's an error, the application works fine.

**What actually happened? Please include the actual, raw output from ESLint.**
``` bash
λ eslint config\auth.js

F:\backend\config\auth.js
  2:9  error  Parsing error: Unexpected token :

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mysticatea** [comment]
Thank you for this issue.

`v2.13.1` is a super old version.
Could you confirm with the latest version?

**Comment by Gelio** [comment]
Upgraded to v3.6.1 and problem is fixed. Thank you for helping me realise that!

I installed eslint globally a long time ago and didn't update it for some reason since then.
